asserts/checkers.go

```diff
@@ -38,12 +39,18 @@ func checkMandatory(headers map[string]string, name string) (string, error) {
 	return value, nil
 }
 
-func checkAssertType(assertType AssertionType) (*assertionTypeRegistration, error) {
-	reg := typeRegistry[assertType]
-	if reg == nil {
-		return nil, fmt.Errorf("unknown assertion type: %v", assertType)
+var errInvalidAssertionType = errors.New("invalid assertion type")
```

Comment:
This is never tested for. Would be more useful to have a fmt.Errorf that reports the actual error.

Also, why is the function below not used in Assemble above when testing for the assertion type given a header?